core/src/ipld.rs

```diff
@@ -16,7 +17,7 @@ pub enum Ipld {
     /// Represents an UTF-8 string.
     String(String),
     /// Represents a sequence of bytes.
-    Bytes(Vec<u8>),
+    Bytes(Box<[u8]>),
```

Comment:
I think vectors are way more common than Boxed slices. So I'm not sure about this change.

Reply:
maybe true, but an immutable opaque blob of bytes should be a `Box<[u8]>` not a `Vec<u8>`.

I mentioned in another PR that `Encode` can only be implemented for `Vec<T>` and not for `Vec<T>` and `Vec<u8>`, and same for `Decode`.